### PR TITLE
Use non-empty lookup data for wasm_invoke fuzz target

### PR DESCRIPTION
**Pull request by rbehjati**
Ref https://github.com/project-oak/oak/pull/2114#issuecomment-846882117

### CODE REVIEW

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs [hunk 1]
```diff
@@ -37,13 +38,21 @@ enum ArbitraryInstruction {
...
+        key: LookupKey,
     },
     WriteLogMessage {
         message: Vec<u8>,
     },
 }
 
+#[derive(Arbitrary, Debug, Clone, PartialEq)]
+enum LookupKey {
+    Key,
```
Comment (tiziano88) [inline oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs:50]: I find it a bit hard to follow this code, basically this enum variant means that the FIXED_KEY value is used, right? Is it just to make sure that there is a balance between the random keys and some key that we think is more useful to find interesting bugs? Or is there some other reason? Maybe also rename `Key` to `DefaultKey` or `FixedKey` and add a comment.

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs [hunk 2]
```diff
@@ -106,11 +119,16 @@ impl From<&ArbitraryInstruction> for crate::proto::Instruction {
                     response: response.clone(),
                 }),
             ),
-            ArbitraryInstruction::StorageGetItem { key } => {
-                Some(InstructionVariant::StorageGetItem(
+            ArbitraryInstruction::StorageGetItem { key } => match key {
+                LookupKey::Key => Some(InstructionVariant::StorageGetItem(
+                    crate::proto::StorageGetItem {
+                        key: FIXED_KEY.to_vec(),
```
Comment (tiziano88) [inline oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs:125]: It looks like the only thing that needs to change between the two match branches is the actual key, but there is no need to repeat all the rest of the expression, right?

File: oak_functions/loader/src/lookup.rs [hunk 3]
```diff
@@ -153,3 +153,24 @@ pub fn parse_lookup_entries<B: prost::bytes::Buf>(
...
+
+/// Wrapper around a LookupData instance for testing.
+#[allow(dead_code)]
+pub struct LookupDataForTest {
+    pub lookup_data: LookupData,
+}
+
+impl LookupDataForTest {
+    /// Creates an instance of LookupDataForTest, containing an instance of LookupData populated
+    /// with the given entires.
```
Comment (tiziano88) [inline oak_functions/loader/src/lookup.rs:165]:
```suggestion
    /// with the given entries.
```

File: oak_functions/loader/src/lookup.rs [hunk 4]
```diff
@@ -153,3 +153,24 @@ pub fn parse_lookup_entries<B: prost::bytes::Buf>(
     }
     Ok(entries)
 }
+
+/// Wrapper around a LookupData instance for testing.
+#[allow(dead_code)]
+pub struct LookupDataForTest {
+    pub lookup_data: LookupData,
+}
```
Comment (tiziano88) [inline oak_functions/loader/src/lookup.rs:161]: AFAICT this wrapper is never actually used, only the inner LookupData instance? If so, why not have a function to create a LookupData instance for test directly?

**Review by tiziano88 — APPROVED** [review]

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs (hunk 1 above)
Comment (rbehjati) [reply oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs]: I am not so much concerned about the balance, but just to make sure that at least once in a while the value `key` is used for lookup, so that we get coverage for the path where lookup hits something in the data. Is there a better way to do this? 

Added clarifying comments and renamed to `FixedKey`.

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs (hunk 2 above)
Comment (rbehjati) [reply oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs]: Fixed.

File: oak_functions/loader/src/lookup.rs (hunk 4 above)
Comment (rbehjati) [reply oak_functions/loader/src/lookup.rs]: True. I was not sure if it is a good idea to add a method for testing to `LookupData`. Fixed.

**Review by rbehjati — COMMENTED** [review]
Thanks :)

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs [hunk 5]
```diff
@@ -107,8 +125,12 @@ impl From<&ArbitraryInstruction> for crate::proto::Instruction {
                 }),
             ),
             ArbitraryInstruction::StorageGetItem { key } => {
+                let key = match key {
+                    LookupKey::FixedKey => FIXED_KEY.to_vec(),
+                    LookupKey::RandomKey { key } => key.clone(),
+                };
```
Comment (tiziano88) [inline oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs:131]: BTW this may also be a method `LookupKey::value()` directly, if you prefer (it seems nicely self contained). This is fine too though.

**Review by tiziano88 — APPROVED** [review]

File: oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs (hunk 5 above)
Comment (rbehjati) [reply oak_functions/loader/fuzz/fuzz_targets/wasm_invoke.rs]: This is nicer. Added a `From` implementation.
